src/components/ui/QuantitySelector/QuantitySelector.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { useState } from 'react'
+import { QuantitySelector as UIQuantitySelector } from '@faststore/ui'
+import { Plus, Minus } from 'phosphor-react'
+
+import './quantity-selector.scss'
+
+interface QuantitySelectorProps {
+  maxQuantity: number
+  minQuantity: number
+  inputDisabled: boolean
+}
+
+export function QuantitySelector({
+  maxQuantity,
+  minQuantity,
+  inputDisabled,
+}: QuantitySelectorProps) {
+  const [quantity, setQuantity] = useState<number>(minQuantity)
+
+  function increase() {
+    setQuantity((currentQuantity) =>
+      validateQuantityBounds(currentQuantity + 1)
+    )
+  }
+
+  function decrease() {
+    setQuantity((currentQuantity) =>
+      validateQuantityBounds(currentQuantity - 1)
+    )
+  }
+
+  function isLeftDisabled(): boolean {
+    return quantity === minQuantity
+  }
+
+  function isRightDisabled(): boolean {
+    return quantity === maxQuantity
+  }
+
+  function validateQuantityBounds(n: number): number {
+    return Math.min(Math.max(n, minQuantity), maxQuantity)
+  }
+
+  function validateInput(e: React.FormEvent<HTMLInputElement>) {
+    const val = e.currentTarget.value
+
+    if (!Number.isNaN(Number(val))) {
+      setQuantity(validateQuantityBounds(Number(val)))
+    }
+  }
+
+  return (
+    <UIQuantitySelector
+      quantity={quantity}
+      className={inputDisabled ? 'disabled' : ''}
```

Comment:
Let's use `data-store-quantity-selector={disabled ? 'disabled' : ''}` instead of class here, ok?

to style it, use:

`[data-store-quantity-selector="disabled"] {}`

Reply:
Sure! Done at https://github.com/vtex-sites/base.store/pull/183/commits/84e0d7538d661bce060d500d12170b9df7f2a50d